src/java.base/share/classes/java/util/random/package-info.java

```diff
@@ -0,0 +1,595 @@
+ * provide trade-offs among speed, space, period, accidental correlation, and
+ * equidistribution properties.
+ *
+ * <p> For applications with no special requirements,
+ * "L64X128MixRandom" has a good balance among speed, space,
+ * and period, and is suitable for both single-threaded and multi-threaded
+ * applications when used properly (a separate instance for each thread).
+ *
+ * <p> If the application uses only a single thread, then
+ * "Xoroshiro128PlusPlus" is even smaller and faster, and
+ * certainly has a sufficiently long period.
+ *
+ * <p> For an application running in a 32-bit hardware environment and using
+ * only one thread or a small number of threads, "L32X64MixRandom" may be a good
+ * choice.
+ *
+ * <p> For an application that uses many threads that are allocated in one batch
+ * at the start of the computation, either a "jumpable" generator such as
+ * "Xoroshiro128PlusPlus" or
+ * "Xoshiro256PlusPlus" may be used, or a "splittable"
+ * generator such as "L64X128MixRandom" or
+ * "L64X256MixRandom" may be used.
+ *
+ * <p> For an application that creates many threads dynamically, perhaps through
+ * the use of spliterators, a "splittable" generator such as
+ * "L64X128MixRandom" or "L64X256MixRandom" is
+ * recommended. If the number of generators created dynamically may
+ * be very large (millions or more), then using generators such as
+ * "L128X128MixRandom" or "L128X256MixRandom",
+ * which use a 128-bit parameter rather than a 64-bit parameter for their LCG
+ * subgenerator, will make it much less likely that two instances use the same
+ * state cycle.
+ *
+ * <p> For an application that uses tuples of consecutively generated values, it
+ * may be desirable to use a generator that is <i>k</i>-equidistributed such
+ * that <i>k</i> is at least as large as the length of the tuples being
+ * generated. The generator "L64X256MixRandom" is provably
+ * 4-equidistributed, and "L64X1024MixRandom" is provably
+ * 16-equidistributed.
+ *
+ * <p> For applications that generate large permutations, it may be best to use
+ * a generator whose period is much larger than the total number of possible
+ * permutations; otherwise it will be impossible to generate some of the
+ * intended permutations. For example, if the goal is to shuffle a deck of 52
+ * cards, the number of possible permutations is 52! (52 factorial), \ which is
+ * larger than 2<sup>225</sup> (but smaller than 2<sup>226</sup>), so it may be
+ * best to use a generator whose period at least 2<sup>256</sup>, such as
+ * "L64X256MixRandom" or "L64X1024MixRandom"
+ * or "L128X256MixRandom" or
+ * "L128X1024MixRandom". (It is of course also necessary to
+ * provide sufficiently many seed bits when the generator is initialized, or
+ * else it will still be impossible to generate some of the intended
+ * permutations.)
+ *
+ *
+ * <h2><a id="algorithms">Random Number Generator Algorithms Available in Java SE</a></h2>
```

Comment:
Some comments and questions on the spec status of this table: is the intentional to require all of these algorithms in all compliant implementation of Java SE or just in the JDK reference implementation? Is the list intended to be exhaustive, meaning no other algorithms should be findable?

I recommend clarifying the intended Java SE vs JDK status here. Also, I recommend including wording along the lines of "this list may change in the future" and "an implementation may provide additional algorithms" etc.

Also, to aid future evolution of the set of algorithms, was there consideration of an "isDeprecated" predicate so that algorithms could be so-marked for a while before being dropped?

Reply:
Precise advice on wording please. The table reflect algorithms guaranteed findable in JDK 17. Whether they are required in all other deployments is something the CSR review should define. "this list may change in the future" is a good idea.

Not sure "isDeprecated" can be done in a meaningful way.  This would require that users would have to add that filter to every query or have all() not include deprecated algorithms. Better to be prepared to handle not finding the algorithm.